apps/mark-scan/backend/src/custom-paper-handler/state_machine.ts

```diff
@@ -1,267 +1,394 @@
-    for await (const { page, pageCount } of pdfToImages(Buffer.from(pdfData), {
-      scale: 200 / 72,
-    })) {
-      assert(pageCount === 1, `Unexpected page count ${pageCount}`);
-      pages.push(page);
-    }
-    const page = pages[0];
-    assert(page, 'Unexpected undefined page');
-    debug(`pdf to image took ${Date.now() - time} ms`);
-    time = Date.now();
+const ScanningState: StateNodeConfig<
+  Context,
+  any,
+  PaperHandlerStatusEvent,
+  BaseActionObject
+> = {
+  invoke: {
+    id: 'scanAndSave',
+    src: (context) => {
+      return scanAndSave(context.driver);
+    },
+    onDone: {
+      target: 'interpreting',
+      actions: assign({ scannedImagePaths: (_, event) => event.data }),
+    },
+  },
+};
 
-    const ballotBinaryBitmap = imageDataToBinaryBitmap(page, options);
-    debug(`bitmap width: ${ballotBinaryBitmap.width}`);
-    debug(`bitmap height: ${ballotBinaryBitmap.height}`);
-    debug(`image to binary took ${Date.now() - time} ms`);
-    time = Date.now();
+const InterpretingState: StateNodeConfig<
+  Context,
+  any,
+  PaperHandlerStatusEvent,
+  BaseActionObject
+> = {
+  invoke: [
+    { ...pollPaperStatus() },
+    {
+      id: 'interpretScannedBallot',
+      src: (context) => {
+        const { scannedImagePaths, workspace } = context;
+        const { store } = workspace;
+        const electionDefinition = store.getElectionDefinition();
 
-    const customChunkedBitmaps = chunkBinaryBitmap(ballotBinaryBitmap);
-    debug(`num chunk rows: ${customChunkedBitmaps.length}`);
-    debug(`binary to chunks took ${Date.now() - time} ms`);
-    time = Date.now();
+        assert(scannedImagePaths);
+        assert(electionDefinition);
 
-    await enablePrintPromise;
-    let dotsSkipped = 0;
-    debug(`begin printing ${customChunkedBitmaps.length} chunks`);
-    let i = 0;
-    for (const customChunkedBitmap of customChunkedBitmaps) {
-      debug(`printing chunk ${i}`);
-      if (customChunkedBitmap.empty) {
-        dotsSkipped += VERTICAL_DOTS_IN_CHUNK;
-      } else {
-        if (dotsSkipped) {
-          await this.driver.setRelativeVerticalPrintPosition(dotsSkipped * 2); // assuming default vertical units, 1 / 408 units
-          dotsSkipped = 0;
-        }
-        await this.driver.printChunk(customChunkedBitmap);
-      }
-      i += 1;
-    }
-    debug('done printing ballot');
-  }
-}
+        const sheetId = uuid();
+        const { precincts } = electionDefinition.election;
+        // Hard coded for now because we don't store precinct in backend. This
+        // will be replaced with a store read in a future PR.
+        const precinct = precincts[precincts.length - 1];
+        const precinctSelection = singlePrecinctSelectionFor(precinct.id);
+        const testMode = true;
+        return interpretScannedBallots(
+          electionDefinition,
+          precinctSelection,
+          testMode,
+          scannedImagePaths,
+          sheetId
+        );
```

Comment:
Nit: maybe factor this out into a helper function to keep the state machine more concise